File: Bindings~/java/src/main/java/com/alttester/Commands/AltCommands/AltSetServerLogging.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester.Commands.AltCommands;

import com.alttester.Commands.AltBaseCommand;
import com.alttester.IMessageHandler;

public class AltSetServerLogging extends AltBaseCommand {

  private AltSetServerLoggingParams setServerLoggingParameters;

  public AltSetServerLogging(
      IMessageHandler messageHandler, AltSetServerLoggingParams setServerLoggingParameters) {
    super(messageHandler);

    this.setServerLoggingParameters = setServerLoggingParameters;
  }

  public void Execute() {
    SendCommand(setServerLoggingParameters);
    recvall(setServerLoggingParameters, String.class);
  }
}

File: Assets/AltTester/Runtime/AltDriver/Commands/AltCommands/AltSetServerLogging.cs
/*
    Copyright(C) 2026 Altom Consulting
*/

using AltTester.AltTesterSDK.Driver.Logging;

namespace AltTester.AltTesterSDK.Driver.Commands
{
    public class AltSetServerLogging : AltBaseCommand
    {
        private readonly AltSetServerLoggingParams cmdParams;

        public AltSetServerLogging(IDriverCommunication commHandler, AltLogger logger, AltLogLevel logLevel) : base(commHandler)
        {
            this.cmdParams = new AltSetServerLoggingParams(logger, logLevel);
        }
        public void Execute()
        {
            this.CommHandler.Send(this.cmdParams);
            var data = this.CommHandler.Recvall<string>(this.cmdParams);
            ValidateResponse("Ok", data);
        }
    }
}
